runtime/config/helpers_internal.go

```diff
@@ -18,18 +18,67 @@ func envName(serviceName string) string {
 	return fmt.Sprintf("ENCORE_CFG_%s", serviceName)
 }
 
-func CreateValue[T any](value T, pathToValue []string) Value[T] {
+// CreateValue creates a new Value on the given path with the given value
+func CreateValue[T any](value T, pathToValue ValuePath) Value[T] {
+	valueID := Singleton.nextID()
 	return func() T {
-		return value
+		Singleton.valueMeta(valueID, pathToValue)
+		return testOverrideOrValue(valueID, value)
 	}
 }
 
-func CreateValueList[T any](value []T, pathToValue []string) Values[T] {
+// CreateValueList creates a new Value Slice on the given path with the given values
+func CreateValueList[T any](value []T, pathToValue ValuePath) Values[T] {
+	valueID := Singleton.nextID()
 	return func() []T {
-		return value
+		Singleton.valueMeta(valueID, pathToValue)
+		return testOverrideOrValue(valueID, value)
 	}
 }
 
+// GetMetaForValue returns the ValueID and ValuePath for the given Value
+func GetMetaForValue[T any](value func() T) (ValueID, ValuePath) {
+	// Get the current request
+	req := Singleton.rt.Current()
+
+	// Lock so we're the only running extraction
+	Singleton.extraction.mutex.Lock()
```

Comment:
Defer unlock right after this might be clearer; I had to scan a bunch to make sure we were unlocking it correctly and it made me a bit nervous